react/src/components/graphs/BarChartCenterAxis.tsx

```diff
@@ -0,0 +1,239 @@
+    dataXr: Array<IXY>;
+    dataXl: Array<IXY>;
+    width: number;
+    height: number;
+    background: string;
+    colorBarLeft: string;
+    colorBarRight: string;
+    settings?: {
+      hideZeroY?: boolean;
+      hideZeroX?: boolean;
+    };
+  };
+}
+
+const defaultSettings = {
+  hideZeroY: false,
+  hideZeroX: false,
+};
+
+const marginTop = 20;
+const marginLeft = 40;
+const marginRight = 40;
+const marginBottom = 40;
+
+const labelProps = {
+  fontFamily: "Open Sans",
+  fontSize: 12,
+};
+
+export default function BarChartCenterAxis(chart: IBarChartCenterAxis) {
+  const fields = { ...chart.fields };
+
+  const {
+    showTooltip,
+    hideTooltip,
+    tooltipOpen,
+    tooltipData,
+    tooltipLeft = 0,
+    tooltipTop = 0,
+  } = useTooltip<TooltipData>({
+    // initial tooltip state
+    tooltipOpen: false,
+    tooltipLeft: fields.width / 3,
+    tooltipTop: fields.height / 3,
+    tooltipData: "",
+  });
+
+  let tooltipTimeout: number;
+
+  if (!fields.settings) {
+    fields.settings = {};
+  }
+  const settings = {
+    ...defaultSettings,
+    ...fields.settings,
+  };
+
+  const chartWidth = fields.width - (marginLeft + marginRight);
+  const chartHight = fields.height - (marginTop + marginBottom);
+
+  const halfWidth = chartWidth / 2;
+
+  const minX = 0;
+  const maxX = Math.max(...fields.dataXr.map((e) => e.x), ...fields.dataXl.map((e) => e.x)) * 1.05;
+
+  const minY = Math.min(...fields.dataY) - 1;
+  const maxY = Math.max(...fields.dataY);
+
+  let barHight = chartHight / fields.dataY.length;
+  if (barHight > 10) {
+    barHight *= 0.7;
+  } else {
+    barHight *= 0.6;
+  }
+
+  const scaleXLeft = scaleLinear({
+    domain: [minX, maxX],
+    range: [halfWidth, 0],
+    round: true,
+    nice: true,
+  });
+
+  const scaleXRight = scaleLinear({
+    domain: [minX, maxX],
+    range: [halfWidth, chartWidth],
+    nice: true,
+    round: true,
+  });
+
+  const scaleY = scaleLinear({
+    domain: [minY, maxY],
+    range: [chartHight, marginTop],
+    nice: true,
+    round: true,
+  });
+
+  return (
+    <>
+      <svg width={fields.width} height={fields.height} style={{ fontFamily: "Open Sans" }}>
+        <rect fill={fields.background} width={fields.width} height={fields.height} />
+        <Group top={marginTop} left={marginLeft}>
+          <Group>
+            <Grid width={chartWidth / 2} height={chartHight} xScale={scaleXLeft} yScale={scaleY} />
+            <Grid
+              left={chartWidth / 2}
+              width={chartWidth / 2}
+              height={chartHight}
+              xScale={scaleXLeft}
+              yScale={scaleY}
+            />
+          </Group>
+          <Group>
+            {fields.dataXl.map((element) => {
+              const barWidth = halfWidth - scaleXLeft(element.x);
+              const x = scaleXLeft(minX) - barWidth;
+              const y = scaleY(element.y);
+              const tooltip = `${fields.labelY} ${element.y}, ${fields.labelXl} ${element.x}`;
```

Comment:
When building locally I get an eslint warning here: `'tooltipData' is already declared in the upper scope on line 68`, same for line 178

Reply:
Nvm, latest commits resolved this :+1: